### ISSUE TITLE
Make possible to use pylance with a local fork of pyright

### ISSUE THREAD

**Issue by savannahostrowski** [microsoft/pylance-release#33]
_Originally posted by @r3m0t in https://github.com/microsoft/pylance-release/issues/4#issuecomment-652922002_

**Comment by erictraut** [comment]
@r3m0t, can you provide more details about why you want to use pylance with a local fork of pyright? What is the use case you have in mind? This can help us explore potential solutions that meet your needs.

**Comment by r3m0t** [comment]
Hi @erictraut, my work uses a monorepo including Python and TypeScript. As our Windows was too old for VFSForGit, a Visual Studio extension handles the files rather than checking them all out to disk. Our tsserver is also customised to read this monorepo, with some js code. As pyright is implemented in js, this could be used with pyright as well.

Another use case is teaching it about some objects in ways that .pyi doesn't support but I haven't looked into this yet.

**Comment by erictraut** [comment]
You're certain welcome to clone the Pyright sources, modify them, and build a private version of the Pyright VS Code Extension or CLI for your own purposes. You won't be able to make use of the additional features provided by Pylance though.

**Comment by archfear** [comment]
On a related note, are there plans for Pylance use released versions Pyright rather than a custom fork. We use Pyright on our CI server and it's import that type checking in VS Code and on the CI server behave identically.

**Comment by erictraut** [comment]
Pylance doesn't use a custom fork of Pyright. The release cadence of Pyright and Pylance are not always 100% sync'ed, so if you see any minor differences between the two, that's probably the explanation. There are also a few small differences in the default configuration settings for the two. Most importantly, "typeCheckingMode" defaults to "basic" in Pyright and "off" in Pylance.

**Comment by archfear** [comment]
@erictraut Thanks for the explanation. Are the defaults for `useLibraryCodeForTypes` different as well? It seemed to be behaving that way. It would be helpful to have the differences in defaults documented so that they can be explicitly set in `pyrightconfig.json` to ensure consistent behavior.

**Comment by erictraut** [comment]
Yes, that's the other default that differs between the two. I think those are the only two.

**Comment by jakebailey** [comment]
Yes, that's should be all of them. The only other may have been workspace vs open file only mode, but IIRC pyright switched to open file mode a while ago, and none of that matters for the CLI.
